masonry_core/src/passes/event.rs

```diff
@@ -207,6 +209,13 @@ pub(crate) fn run_on_pointer_event_pass(root: &mut RenderRoot, event: &PointerEv
         }
     }
 
+    if let Some(id) = target_widget_id {
+        if !root.widget_arena.has(id) {
+            debug_panic!("Trying to target non-existent widget {id}");
+            return Handled::No;
+        }
+    }
```

Comment:
How is this different to the check which happens at the very start of `run_event_pass`? If it is meaningfully so, why isn't this copied to the other callers?